core/src/test/java/org/elasticsearch/common/settings/loader/JsonSettingsLoaderTests.java

```diff
@@ -48,6 +49,10 @@ public void testSimpleJsonSettings() throws Exception {
     }
 
     public void testDuplicateKeysThrowsException() {
+        if (JsonXContent.isStrictDuplicateDetectionEnabled()) {
```

Comment:
What about `assumeTrue("Test only makes sense if json parser doesn't have strict duplicate checks enabled and can be removed if we force it to be enabled", JsonXContent.isStrictDuplicateDetectionEnabled());`?